### DISCUSSION TITLE
Is Tez executions supported

### DISCUSSION THREAD

**Discussion by ChristopherBaldwin** [beltran/gohive#173] [Q&A] [answered]
I'm successfully able to use _gohive_ for queries that return a single result set but I can't get it to work when the query is a Tez job.  It returns after the first vertex succeeds and not when the entire job is complete.  Is there a setting that I'm missing?

**Comment by beltran** [comment]
This shouldn't be the case,  to clarify, what's happening is that `cursor.Exec` returns, `cursor.HasMore` is false, `cursor.Err` is nil and the has a single row, right? By any chance did you try with other clients and got a different result? Could you try to [enable logging](https://github.com/beltran/gohive/blob/cb0d49e69d68a27c1f16f7bd217125c799de0a43/hive.go#L441)?

**Comment by ChristopherBaldwin** [comment, marked as answer]
`cursor.Exec` returns, `cursor.HasMore` is false, and `cursor.Err` has the first TEZ result:

```
[{"returnValue":1,"beginTime":1644252458639,"endTime":1644252458846,"taskId":"Stage-1","taskState":"FINISHED","taskType":"MAPRED","name":"TEZ","elapsedTime":207}]
```

I setup logging and sure enough found the issue.  It was a permissions failure.  I saw the above `FINISHED` but didn't notice the `returnValue`.  Posting this so that others don't make the same mistake.

@beltran Thanks for the fast response.  Sorry.  My bad.

**Comment by beltran** [comment]
Happy to see it's solved, thank you for following up!
